src/dexter/logger.cr

```diff
@@ -1,28 +1,29 @@
 require "logger"
 require "./formatters/*"
 
-class Dexter::Logger < Logger
-  property log_formatter : Dexter::Formatters::BaseLogFormatter.class
+module Dexter
+  class Logger < ::Logger
+    property log_formatter : Dexter::Formatters::BaseLogFormatter.class
 
-  # The built-in Crystal Logger requires a formatter, but we don't use it.
-  # We instead override the `write` method and use our own formatter that
-  # accepts a NamedTuple instead of a string
-  private UNUSED_FORMATTER = Formatter.new do |_, _, _, _, _|
-    # unused
-  end
+    # The built-in Crystal Logger requires a formatter, but we don't use it.
+    # We instead override the `write` method and use our own formatter that
+    # accepts a NamedTuple instead of a string
+    private UNUSED_FORMATTER = Formatter.new do |_, _, _, _, _|
+      # unused
+    end
 
-  def initialize(
-    @io : IO?,
-    @level = Severity::INFO,
-    @log_formatter = Dexter::Formatters::JsonLogFormatter,
-    @progname = ""
-  )
-    @formatter = UNUSED_FORMATTER
-    @closed = false
-    @mutex = Mutex.new
-  end
+    def initialize(
+      @io : IO?,
+      @level = Severity::INFO,
+      @log_formatter = Dexter::Formatters::JsonLogFormatter,
+      @progname = ""
+    )
+      @formatter = UNUSED_FORMATTER
```

Comment:
You can (and IMHO should) call `super` here.

Reply:
I'll look into it. Thanks for the advice